src/cascadia/TerminalApp/IAltListener.idl

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace TerminalApp
+{
+    // C++/winrt makes it difficult to share this idl between two projects,
+    // Instead, we just pin the uuid and include it in both TermControl and App
+    // If you update this one, please update the one in TerminalControl\TermControl.idl
+    // If you change this interface, please update the guid.
+    // If you press Alt and get a runtime error, go make sure both copies are the same.
+    [uuid("36a6554d-07e9-4861-8d8b-5d882de31e75")]
+    interface IAltListener {
+        Boolean OnAltReleased();

```

Comment:
Just to nip this in the bud for the future. I'm sure we'll get one or two more of these, we just haven't figured out what they are yet.

So that end, should we generalize the interface to be `IDirectKeyListener` and give it a method whose signature is like...

```
Boolean OnGotAKey(Uint32 vkey, Boolean down);
```

?

It'll let us consolidate the code in TermControl and maybe clean up the dispatcher in AppHost/the main event loop as well.

Reply:
Considering I've got time today, yea might as well.